sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -73,48 +74,44 @@ public static int calculateBitSetWidthInBytes(int numFields) {
   /**
    * Field types that can be updated in place in UnsafeRows (e.g. we support set() for these types)
    */
-  public static final Set<DataType> mutableFieldTypes;
+  public static final Set<PhysicalDataType> mutableFieldTypes;
 
   // DecimalType, DayTimeIntervalType and YearMonthIntervalType are also mutable
   static {
     mutableFieldTypes = Collections.unmodifiableSet(
       new HashSet<>(
         Arrays.asList(
-          NullType,
-          BooleanType,
-          ByteType,
-          ShortType,
-          IntegerType,
-          LongType,
-          FloatType,
-          DoubleType,
-          DateType,
-          TimestampType,
-          TimestampNTZType
+          PhysicalNullType$.MODULE$,
```

Comment:
this looks ugly. Instead of having a Set, shall we just add a `def isPrimitive` in `PhysicalDataType`?

Reply:
Thank you, I will try to modify the code as you say



> this looks ugly. Instead of having a Set, shall we just add a `def isPrimitive` in `PhysicalDataType`?
Thank you, I will try to modify the code as you say

Reply:
> this looks ugly. Instead of having a Set, shall we just add a `def isPrimitive` in `PhysicalDataType`?
done